### ISSUE TITLE
I can not get `accesslog`  directly from `rsyslog`

### ISSUE THREAD

**Issue by wkingfly** [benoitc/gunicorn#1099]
This stop me at glogging.Logger.access 
```
if not self.cfg.accesslog and not self.cfg.logconfig:
    return
```

**Comment by tilgovi** [comment]
I'm sorry, I don't understand your issue. Where is this code?

**Comment by benoitc** [comment]
@wkingfly bump

**Comment by benoitc** [comment]
@wkingfly Actually access logs are not supposed to be sent in syslog. Is this what you're asking for?

**Comment by wkingfly** [comment]
@benotic I just wanted to unify management of all logs to filter classification. But why not let access logs to syslog

**Comment by wkingfly** [comment]
@tilgovi The code in gunicorn/glogging.py Logger.access

**Comment by benoitc** [comment]
@wkingfly so I checked at the code. 

It is supposed to [be handled](https://github.com/benoitc/gunicorn/blob/master/gunicorn/glogging.py#L198-L200) but indeed [the check](https://github.com/benoitc/gunicorn/blob/master/gunicorn/glogging.py#L275-L276) you were referring prevent it.

It comes from the fact that access logs should be enabled/or disabled on demand. It is actually disabled by default. To do it we are right now only checking if the output is given in setting using the `accesslog`property or the command line `--access-logfile`.

Not sure what is the right way to do it. Either we could have add a new setting `--log-access-to-syslog` ( shorter name would be cool) or introduce a new setting `--log-access-to={syslog,file,none}`. Thoughts?

Fix can be linked with #107O imo.

**Comment by wkingfly** [comment]
@benoitc I think maybe add a new setting `--enable-accesslog`, when add it, default it will go to syslog, how do you think this?

**Comment by benoitc** [comment]
the default can't be syslog for sure. In my understanding the access log will always be enable by default like the error logs right now. But we may  need to distinct which one we want to silent. This without adding too much options. I already think we have too much options :)

**Comment by wkingfly** [comment]
@benoitc yeah, agreed, default error logs through rsyslog, so default access log through rsyslog too, yes?

**Comment by tilgovi** [comment]
I think the access log defaults to off because it's noisy in the console if you're just trying to see error logs on stderr. I think we've talked about access log on stdout, but maybe decided that was too noisy for the default. But maybe we should reconsider, at least for non-stdio logs, that maybe we log access by default and let the logging system decide what to include or exclude.

**Comment by sebest** [comment]
Hi,

It seems that the current behaviour (as of 19.3) when we enable syslog is to also send accesslog even if --access-logfile is None (default value). I also tried setting it to /dev/null.

Is there a way to not log access logs while still enabling syslog for errors?

thanx

**Comment by wkingfly** [comment]
@sebest It already supports, hope the following code to help you:

```
class Logger(object):
    ...
    ...
    def setup(self, cfg):
        ...
        ...
        # set gunicorn.error handler
        self._set_handler(self.error_log, cfg.errorlog, logging.Formatter(self.error_fmt, self.datefmt))
```

from gunicorn/glogging.py

**Comment by sebest** [comment]
@wking i was hoping for a cmd-line option.

The code that you copy paste is for the error log? I want to disable the access logs going to syslog.

**Comment by sebest** [comment]
from what i read here:
https://github.com/benoitc/gunicorn/blob/master/gunicorn/glogging.py#L194

As soon as you enable syslog both errors and access logs will be forwarded to syslog, but i want only the error logs as i logging access log with nginx already.

**Comment by benoitc** [comment]
@tilgovi i guess it's fixed right?

**Comment by tilgovi** [comment]
Yes but I didn't close this in case it doesn't fully address what's needed.
I think it should, though.

**Comment by tilgovi** [comment]
@benoitc oh yes, I thought you were talking about the other issue where I made that change.
Fixed in 2f8e7503fed67dcf717017b012faabc36b46da52 and released in 19.4.3.
